### PR TITLE
Update non_null.ex with details about adding and removing

### PR DESCRIPTION
**Pull request by alanvardy**
Hello and thank you for the project.

I was confused around adding and removing non_null now that the correctness is more strictly enforced with 1.7.0. I received clarification from @benwilson512 and hope that this edit helps. I can reword this if anyone can think of a better way of putting it.

### CODE REVIEW

File: lib/absinthe/type/non_null.ex [hunk 1]
```diff
@@ -5,6 +5,8 @@ defmodule Absinthe.Type.NonNull do
 
   By default, all types in GraphQL are nullable. To declare a type that
   disallows null, wrap it in a `Absinthe.Type.NonNull` struct.
+  
+  Adding non_null/1 to a type is a breaking change, removing is not.
```
Comment (benwilson512) [inline lib/absinthe/type/non_null.ex:9]:
```suggestion
  Adding non_null/1 to a type is a breaking change, removing it is not. Client documents that specify non null on a 
  variable eg `query ($id: ID!)` are allowed to be passed to arguments which allow null. If the argument however is
  non_null, then the variable type MUST be non null as well.
```

**Comment by benwilson512** [comment]
Thank you!

**Comment by maartenvanvliet** [comment]
There's a bit of nuance here. 

Adding non_null/1 to an **input** type is a breaking change, whereas adding non_null/1 to an **output** type is not. 
The inverse also holds, removing non_null from an **input** type is non-breaking, removing non_null/1 from an **output** type is breaking.

**Comment by benwilson512** [comment]
@maartenvanvliet ah yes good point, I was coming at it from the angle of inputs given the recent bug fix. I'll make an update to handle both situations.

**Comment by alanvardy** [comment]
Very interesting, thank you @maartenvanvliet and @benwilson512 !

**Pull request merged**